components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java

```diff
@@ -171,6 +171,12 @@ public class ApplicationMgtDBQueries {
             + "WHERE INBOUND_AUTH_KEY = ? AND INBOUND_AUTH_TYPE = ? AND SP_APP.TENANT_ID = ? "
             + "AND SP_INBOUND_AUTH.TENANT_ID=?";
 
+    public static final String LOAD_APP_UUID_BY_CLIENT_ID = "SELECT UUID "
+            + "FROM SP_APP INNER JOIN SP_INBOUND_AUTH "
+            + "ON SP_APP.ID = SP_INBOUND_AUTH.APP_ID "
+            + "WHERE INBOUND_AUTH_KEY = ? AND INBOUND_AUTH_TYPE = ? AND SP_APP.TENANT_ID = ? "
+            + "AND SP_INBOUND_AUTH.TENANT_ID=?";
```

Comment:
Let's confirm this hits indexes. Shouldn't be a problem though as LOAD_APPLICATION_NAME_BY_CLIENT_ID_AND_TYPE uses similar columns